src/core/bonded_interactions/angle_common.hpp

```diff
@@ -0,0 +1,98 @@
+/*
+  Copyright (C) 2010-2019 The ESPResSo project
+  Copyright (C) 2002-2010
+    Max-Planck-Institute for Polymer Research, Theory Group
+
+  This file is part of ESPResSo.
+
+  ESPResSo is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  ESPResSo is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#ifndef ANGLE_COMMON_H
+#define ANGLE_COMMON_H
+/** \file
+ *  Common code for functions calculating angle forces.
+ */
+
+#include "grid.hpp"
+#include "particle_data.hpp"
+#include "utils.hpp"
+
+/** Compute a three-body angle interaction force.
+ *  @param[in]  p_mid        Second/middle particle.
+ *  @param[in]  p_left       First/left particle.
+ *  @param[in]  p_right      Third/right particle.
+ *  @param[in]  forceFactor  Angle bending constant.
+ *  @param[out] force1       Force on particle 1.
+ *  @param[out] force2       Force on particle 2.
+ */
+template <typename ForceFactor>
+void calc_angle_generic_force(Particle const *p_mid, Particle const *p_left,
```

Comment:
You can narrow the interface here: only the positions are needed, not the particles themselves.

Reply:
Done!